src/JustEat.StatsD/StatsDPublisher.cs

```diff
@@ -129,9 +133,30 @@ public void MarkEvent(string name)
 
         private void Send(string metric)
         {
+            if (string.IsNullOrEmpty(metric))
+            {
+                return;
+            }
+
             try
             {
-                _transport.Send(metric);
+#if NETCOREAPP2_1
+                const int MaxStackallockThresholdInBytes = 256;
+                const int MaxUTF16CharsToFitOnStack = MaxStackallockThresholdInBytes / 2;
+                if (metric.Length <= MaxUTF16CharsToFitOnStack)
+                {
+                    int maxBytes = metric.Length * 2;
+                    Span<byte> destination = stackalloc byte[maxBytes];
+                    var written = Encoding.UTF8.GetBytes(metric, destination);
+                    _transport.Send(destination.Slice(0, written));
+                }
+                else
+                {
+                    _transport.Send(Encoding.UTF8.GetBytes(metric));
+                }
+#else
+                _transport.Send(Encoding.UTF8.GetBytes(metric));
```

Comment:
This could come out of the `#else` block if your were to return on like 152, then you could get rid of the duplicated call above.